be/src/block_cache/fb_cachelib.cpp

```diff
@@ -45,31 +45,20 @@ Status FbCacheLib::write_cache(const std::string& key, const char* value, size_t
     return Status::OK();
 }
 
-StatusOr<size_t> FbCacheLib::read_cache(const std::string& key, char* value) {
+StatusOr<size_t> FbCacheLib::read_cache(const std::string& key, char* value, size_t off, size_t size) {
     // TODO:
     // 1. check chain item
     // 2. replace with async methods
     auto handle = _cache->find(key);
     if (!handle) {
         return Status::NotFound("not found cachelib item");
     }
-    size_t size = handle->getSize();
-    std::memcpy(value, handle->getMemory(), size);
+    std::memcpy(value, (char*)handle->getMemory() + off, size);
```

Comment:
If size is larger than handle->getSize(), it will cause memory access error?

Reply:
yes, but application level will ensure [off, off + size] is in memory range.